pkg/efivarfs/varfs.go

```diff
@@ -0,0 +1,243 @@
+// Copyright 2014-2021 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+//
+// This code is a somewhat simplified and changed up version of
+// github.com/canonical/go-efilib credits go their authors.
+// It allows interaction with the efivarfs via u-root which means
+// both read and write support.
+
+package efivarfs
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"sort"
+	"syscall"
+
+	"github.com/u-root/u-root/pkg/guid"
+	"golang.org/x/sys/unix"
+)
+
+func init() {
+	if !probeEfivarfs() {
+		return
+	}
+}
+
+// EfiVarFs is the path to the efivarfs mount point
+const EfiVarFs = "/sys/firmware/efi/efivars/"
+
+var vars varsBackend = varfsBackend{}
+
+type varsBackend interface {
+	Get(name string, guid guid.GUID) (VariableAttributes, []byte, error)
+	Set(name string, guid guid.GUID, attrs VariableAttributes, data []byte) (bool, error)
+	Delete(name string, guid guid.GUID) error
+	List() ([]VariableDescriptor, error)
+}
+
+type varfsBackend struct{}
```

Comment:
you can just call it Backend, since varfs is part of the package name (efivarfs)